### PR TITLE
catch and convert all aio_pika connection exceptions

### LINKED ISSUE #8544: PikaEventBroker: ConnectionError: [Errno -2] Name or service not known
Sentry Issue: [RASA-OPEN-SOURCE-39Y](https://sentry.io/organizations/rasahq/issues/2093479575/?referrer=github_integration)

```
gaierror: [Errno -2] Name or service not known
  File "aiormq/connection.py", line 229, in connect
    self.url.host, self.url.port, ssl=ssl_context,
  File "asyncio/streams.py", line 77, in open_connection
    lambda: protocol, host, port, **kwds)
  File "uvloop/loop.pyx", line 1914, in create_connection

ConnectionError: [Errno -2] Name or service not known
(22 additional frame(s) were not displayed)
...
  File "rasa/core/brokers/broker.py", line 74, in _create_from_endpoint_config
    broker = await PikaEventBroker.from_endpoint_config(endpoint_config, event_loop)
  File "rasa/core/brokers/pika.py", line 138, in from_endpoint_config
    await broker.connect()
  File "rasa/core/brokers/pika.py", line 144, in connect
    self._connection = await self._connect()
  File "rasa/core/brokers/pika.py", line 188, in _connect
    raise last_exception
  File "rasa/core/brokers/pika.py", line 174, in _connect
    ssl_options=ssl_options,
```

**Definition of Done**
- [x] catch and convert to subclass of `RasaException`
- [x] catch and convert all `CONNECTION_EXCEPTIONS` (defined in `air_pika/exceptions.py`) to subclasses of RasaException

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
-  Fix #8544 

**Status (please check what you already did)**:
- [X] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/brokers/broker.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ async def create(
             sqlalchemy.exc.OperationalError,
             aio_pika.exceptions.AMQPConnectionError,
             aiormq.exceptions.ChannelNotFoundEntity,
+            *aio_pika.exceptions.CONNECTION_EXCEPTIONS,
```
Comment (joejuzl) [inline rasa/core/brokers/broker.py:37]: Are some of the errors above included in this?

File: changelog/8544.misc.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Catch and convert all `aio_pika` connection exceptions to subclass of `RasaException`.
```
Comment (joejuzl) [inline changelog/8544.misc.md:1]: Only in event broker creation right?

**Review by joejuzl — CHANGES_REQUESTED** [review]

**Comment by wochinge** [comment]
Can we please also ignore this one https://github.com/RasaHQ/rasa/issues/8112 ?

File: rasa/core/brokers/broker.py (hunk 1 above)
Comment (ancalita) [reply rasa/core/brokers/broker.py:37]: I've just double-checked, the errors above are not included in `aio_pika.exceptions.CONNECTION_EXCEPTIONS`.

File: changelog/8544.misc.md (hunk 2 above)
Comment (ancalita) [reply changelog/8544.misc.md:1]: Yup, that's where I traced the error in the Sentry stacktrace. I'll add this note to the changelog.

**Comment by ancalita** [comment]
@wochinge Yup, the entire built-in `ConnectionError` class is part of `aio_pika.exceptions.CONNECTION_EXCEPTIONS` and should be handled now. Nothing to action here, apart from closing 2 issues in one go after merge? 😄

**Review by joejuzl — APPROVED** [review]

**Comment by wochinge** [comment]
Amazing 🚀

**Pull request merged**